### DISCUSSION TITLE
Expanding relation

### DISCUSSION THREAD

**Discussion by mankinchi** [pocketbase/pocketbase#1910] [Q&A] [answered]
From my research, I think this is impossible, so I'm posting this here hoping someone can prove me wrong. 

Below are my database
<img width="801" alt="image" src="https://user-images.githubusercontent.com/8164063/221338593-85224f4e-c8e7-48ec-ac38-a388b55f4fef.png">

I want to grab the questionnaire, then expand to questions, question.type, question.options,... My current API expand looks like this `?expand=question(questionnaire).type,question(questionnaire).option`, but only the `type` gets executed. 

Am I doing something wrong or is this not possible to go this deep? Thanks everyone 😄

**Comment by ganigeorgiev** [comment, marked as answer]
**There is no `option` relation field in the above** and I'm not sure that I understand what is the expected result.

If you want to expand `QuestionOptions`, then as far as I understand from the above graph it is indirect relation to question, right? If that's the case then your `questionnaire`'s `expand` should look something like:

```
?expand=question(questionnaireId).type,question(questionnaireId).questionOptions(questionId)
```
_(make sure that your request satisfies the "View API" rule of the rel collections)_

**Reply by mankinchi** [reply]
I see what you mean. Initially, I didn't understand that you can nested indirect relations, such as `question(questionnaireId).questionOptions(questionId)`.  Thanks for clarifying 👍
